lhotse/dataset/collation.py

```diff
@@ -168,6 +168,66 @@ def collate_audio(
     return audio, audio_lens
 
 
+def collate_custom_field(
+    cuts: CutSet,
+    field: str,
+    pad_value: Union[None, int, float] = None,
+    pad_direction: str = "right",
+) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.IntTensor]]:
+    """
+    Load custom arrays for all the cuts and return them as a batch in a torch tensor.
+    The output shapes are:
+
+        - ``(batch, d0, d1, d2, ...)`` for :class:`lhotse.array.Array` of shape ``(d0, d1, d2, ...)``.
+            Note: all arrays have to be of the same shape, as we expect these represent fixed-size
+            embeddings.
+
+        - ``(batch, d0, pad_dt, d1, ...)`` for :class:`lhotse.array.TemporalArray` of shape
+            ``(d0, dt, d1, ...)`` where ``dt`` indicates temporal dimension (variable-sized),
+            and ``pad_dt`` indicates temporal dimension after padding (equal-sized for all cuts).
+            We expect these represent temporal data, such as alignments, posteriors, features, etc.
+
+        - ``(batch, )`` for anything else, such as int or float: we will simply stack them into
+            a list and tensorize it.
+
+    :param cuts: a :class:`CutSet` used to load the features.
+    :param field: name of the custom field to be retrieved.
+    :param pad_value: value to be used for padding the temporal arrays.
+        Ignored for non-temporal array and non-array attributes.
+    :param pad_direction: where to apply the padding (``right``, ``left``, or ``both``).
+    :return: a collated data tensor, or a tuple of tensors ``(collated_data, sequence_lens)``.
+    """
+    from lhotse.array import Array, TemporalArray
+
+    first_manifest = getattr(cuts[0], field)
+    if isinstance(first_manifest, Array):
+        # Expected data type: fixed-size embeddings.
+        # Simply stack across a new dimension inserted at 0.
+        assert all(getattr(c, field).shape == first_manifest.shape for c in cuts), (
+            "Cannot collate manifests of type Array with different shapes, "
+            "because we don't know which dimension must be padded. "
+            "Use TemporalArray manifests and try again."
+        )
+        return torch.stack([torch.from_numpy(c.load_custom(field)) for c in cuts])
+    elif isinstance(first_manifest, TemporalArray):
+        # Expected data type: variable-sized tensors (along only one dimension).
+        # Pad across that dimension, then stack at dimension 0.
+        assert (
+            pad_value is not None
```

Comment:
How about reading the padding value from cut's attribute?

For instance, if there is a custom field called `xxx` and if
there exists another field called `xxx_pad_value`, can we use `xxx_pad_value`
as the padding value for `xxx`?

 If there is no `xxx_pad_value`, can we use a default value, say 0, for the padding value?

We are doing similar things in k2, see
https://github.com/k2-fsa/k2/blob/bdcaaf828212913366110a0c45aab4ea40b1cbf9/k2/python/k2/fsa.py#L308
```
        ans = getattr(self, attribute_name + '_filler', 0)
```